api/src/main/java/io/opentelemetry/distributedcontext/DistributedContext.java

```diff
@@ -56,9 +56,9 @@
    */
   interface Builder {
     /**
-     * Sets the parent {@link DistributedContext} to use. If not set, the value of {@link
-     * DistributedContextManager#getCurrentContext()} at {@link #build()} time will be used as
-     * parent.
+     * Sets the parent {@link DistributedContext} to use. If no parent (or {@code null}) is
```

Comment:
Not sure it is clear to me what the behavior should be if null. Probably we should not allow null.

Reply:
With the builder pattern something like "setParent(getMyContext()) but only if it's not null" is a bit cumbersome to write but on the other hand it could be surprising that an implicit parent is used after the user tried setting another one which turned out to be null, yes.
I'm fine either way. What would you prefer, @songy23?

Reply:
IMO we can rephrase the statement here to be "If you want to set an explicit parent, use setParent(). Explicit parent cannot be null. If you want to use the value in current context (which could be null), you don't need to call this method." Then we can enforce `setParent` to take non-null.

Reply:
I missed the `@throws NullPointerException if {@code parent} is {@code null}` declaration we already have in setParent's doc. The current `DistributedContextSdkTest.setParent_nullValue` also asserted a different behavior (clearing previously set parent).
I think this leads to least surprise (no implicit parent if setParent is called with null) so I'll implement it for now - this also assures consistency with `NoopDistributedContextBuilder`. Once https://github.com/open-telemetry/opentelemetry-specification/pull/153 is settled this can be revisited.